### DISCUSSION TITLE
PROBLEM WITH ZIGBEE2MQTT

### DISCUSSION THREAD

**Discussion by charles8119** [Koenkk/zigbee2mqtt#6656] [Devices]
I have a problem when start ZIGBEE2MQTT. I can´t connect my CC2531.

RASPBERRY 4 PI
HA: core-2021.3.3
Zigbee2mqtt: 1.18.1-1

PROBLEM:

Zigbee2MQTT:error 2021-03-10 17:10:44: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-03-10 17:10:44: Failed to start zigbee
Zigbee2MQTT:error 2021-03-10 17:10:44: Exiting...
Zigbee2MQTT:error 2021-03-10 17:10:44: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.<anonymous> (/app/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:94:31)
    at Generator.throw (<anonymous>)
    at rejected (/app/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:25:65)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.18.1 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.18.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-03-10T20_10_44_770Z-debug.log

====================

COnfig:

data_path: /config/zigbee2mqtt
external_converters: []
devices: devices.yaml
groups: groups.yaml
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://localhost'
  user: usermqtt
  password: usermqtt
serial:
  port: /dev/ttyAMA0
advanced:
  log_level: warn
  pan_id: 6754
  channel: 11
  network_key:
    - 1
    - 3
    - 5
    - 7
    - 9
    - 11
    - 13
    - 15
    - 0
    - 2
    - 4
    - 6
    - 8
    - 10
    - 12
    - 13
  availability_blocklist: []
  availability_passlist: []
device_options: {}
blocklist: []
passlist: []
queue: {}
frontend:
  port: 8099
experimental: {}
socat:
  enabled: false
  master: 'pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777'
  slave: 'tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5'
  options: '-d -d'
  log: false

=========================

produto CC2531 dongle USB

idProduct 16ae

idVendor 0451 
busnum 1

dev 189: 5

versão 2.00 
velocidade 12

**Comment by Twit123** [comment]
are you sure your CC2531 is connected to /dev/ttyAMA0?

Supervisor-->System then click the 3 dots at the bottom of the host card and then hardware. You should get a popup listing all hardware. Do a search on /dev/ttyAMA0 and see if it is there.

Simon

**Reply by kd3c** [reply]
I have the same issue after reinstalling the home assistant. I followed all checks recommended in the Zigbee2MQTT FAQ section and various other suggestions. What is really odd is the fact the coordinator doesn't work on the old instance of the home assistant (I still have an old SD card with the instance that worked fine 2 days ago, I reinstalled home assistant using a bigger SD card and I wanted to get HACS working properly and now Zigbee2MQTT doesn't work at all).  
I confirmed device port using `ls -l /dev/serial/by-id` and get expected output as per: https://www.zigbee2mqtt.io/information/FAQ.html
I verified that I have write access. The last thing I want to try is to reflash my USB stick. 
I will appreciate any suggestions.

**Comment by Jim-Noble** [comment]
With the CC2531 I have found that sometimes you need to unplug the stick and re-insert it to get it to work again. I guess something gets into a state a soft reset just can't unravel...

**Reply by kd3c** [reply]
Hi. I tried to reset adapter countless times to no avail.  I can't get my head around this. Why it doesn't work on the old image?
this is my log:

` 2021-03-11 22:18:41: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null},"backupPath":"/config/zigbee2mqtt/coordinator_backup.json","databaseBackupPath":"/config/zigbee2mqtt/database.db.backup","databasePath":"/config/zigbee2mqtt/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":6754},"serialPort":{"path":"/dev/ttyACM0"}}'
2021-03-11T22:18:42.281Z zigbee-herdsman:adapter Failed to validate path: 'Error: spawn udevadm ENOENT'
2021-03-11T22:18:42.285Z zigbee-herdsman:controller:log Starting with options '{"network":{"networkKeyDistribute":false,"networkKey":[1,3,5,7,9,11,13,15,0,2,4,6,8,10,12,13],"panID":6754,"extendedPanID":[221,221,221,221,221,221,221,221],"channelList":[11]},"serialPort":{"path":"/dev/ttyACM0"},"databasePath":"/config/zigbee2mqtt/database.db","databaseBackupPath":"/config/zigbee2mqtt/database.db.backup","backupPath":"/config/zigbee2mqtt/coordinator_backup.json","adapter":{"concurrent":null,"delay":null}}'
2021-03-11T22:18:42.289Z zigbee-herdsman:adapter:zStack:znp:log Opening SerialPort with /dev/ttyACM0 and {"baudRate":115200,"rtscts":false,"autoOpen":false}
2021-03-11T22:18:47.332Z zigbee-herdsman:adapter:zStack:znp:log Serialport opened
2021-03-11T22:19:02.918Z zigbee-herdsman:adapter:zStack:znp:log Writing skip bootloader payload
2021-03-11T22:19:02.920Z zigbee-herdsman:adapter:zStack:unpi:writer --> buffer [239]
2021-03-11T22:19:03.931Z zigbee-herdsman:adapter:zStack:znp:SREQ --> SYS - ping - {"capabilities":1}
2021-03-11T22:19:03.938Z zigbee-herdsman:adapter:zStack:unpi:writer --> frame [254,0,33,1,32]
2021-03-11T22:19:09.948Z zigbee-herdsman:adapter:zStack:znp:SREQ --> SYS - ping - {"capabilities":1}
2021-03-11T22:19:09.949Z zigbee-herdsman:adapter:zStack:unpi:writer --> frame [254,0,33,1,32]
2021-03-11T22:19:15.956Z zigbee-herdsman:adapter:zStack:znp:SREQ --> SYS - ping - {"capabilities":1}
2021-03-11T22:19:15.958Z zigbee-herdsman:adapter:zStack:unpi:writer --> frame [254,0,33,1,32]
Zigbee2MQTT:error 2021-03-11 22:19:21: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-03-11 22:19:21: Failed to start zigbee
Zigbee2MQTT:error 2021-03-11 22:19:21: Exiting...
Zigbee2MQTT:error 2021-03-11 22:19:21: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)`

**Comment by DubhAd** [comment]
`/dev/ttyAMA0` isn't the USB stick - the USB stick would be more likely to be `/dev/ttyACM0` or `/dev/ttyUSB0`

**Comment by charles8119** [comment]
I didn't find this door, but I managed to take the test to write in the folder successfully.

I use this command "test -w /dev/ttyAMA0 && echo success || echo failure"  and I received " sucess".

Can you help me?

nome: 1-1 
sysfs:> - 
/sys/devices/platform/scb/fd500000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0/usb1/1-1 
dev_path: / dev / bus / usb / 001/002 
subsistema: usb 
by_id: 
atributos nulos : 
BUSNUM: '001' 
DEVNAME: / dev / bus / usb / 
001/002 DEVNUM: '002' 
DEVPATH:> - 
/devices/platform/scb/fd500000.pcie/ pci0000: 00/0000: 00: 00.0 / 0000: 01: 00.0 / usb1 / 1-1 
DEVTYPE: usb_device 
DRIVER: usb 
ID_BUS: usb 
ID_FOR_SEAT: usb-platform-fd500000_pcie-pci-0000_01_00_0-usb-0_1 
ID_MODEL: USB2.0_Hub 
ID_MODEL_ENC: USB2.0 \ x20Hub 
ID_MODEL_ID: '3431' 
ID_PATH: 'platform-fd500000.pcie-pci-0000: 01: 00.0-usb-0: 1'
ID_PATH_TAG: plataforma de fd500000_pcie-pci-0000_01_00_0-usb-0_1 
ID_REVISION: '0421' 
ID_SERIAL: 2109_USB2.0_Hub 
ID_USB_INTERFACES: ': 090.000:' 
ID_VENDOR: '2109' 
ID_VENDOR_ENC: '2109' 
ID_VENDOR_ID: '2109' 
MAJOR: '189' 
MENOR: '1' 
PRODUTO: 2109/3431/421 
SUBSISTEMA: usb 
TAGS: 'assento' 
TYPE: 9/0/1 
USEC_INITIALIZED:' 2257738 ' 
** - nome: 1-1.3 
sysfs:> - 
/sys/devices/platform/scb/fd500000.pcie/pci0000:00/0000:00:00.0/0000: 01: 00.0 / usb1 / 1-1 / 
1-1.3 dev_path: / dev / bus / usb / 001/003 
subsistema: usb 
by_id: 
atributos nulos : 
BUSNUM: '001' 
DEVNAME: / dev / bus / usb / 001/003 
DEVNUM: '003' 
DEVPATH:> - 
/devices/platform/scb/fd500000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0/usb1/1-1/1-1.3 
DEVTYPE: usb_device 
DRIVER: usb 
ID_BUS: usb 
ID_FOR_SEAT: usb-platform-fd500000_pcie-pci-0000_01_00_0-usb-0_1_3 
ID_MODEL: CC2531_USB_Dongle 
ID_MODEL_ENC: CC2531 \ x20USB \ x20Dongle
ID_MODEL_ID: 16ae 
ID_PATH: 'plataforma-fd500000.pcie-pci-0000: 01: 00.0-usb-0: 1.3' 
ID_PATH_TAG: plataforma de fd500000_pcie-pci-0000_01_00_0-usb-0_1_3 
ID_REVISION: '1472' 
ID_SERIAL: Texas_Instruments_CC2531_USB_Dongle 
ID_USB_INTERFACES: ' : ffffff: ' 
ID_VENDOR: Texas_Instruments 
ID_VENDOR_ENC: Texas \ x20Instruments 
ID_VENDOR_ID:' 0451 ' 
PRINCIPAL:' 189 ' 
MENOR:' 2 ' 
PRODUTO: 451 / 16ae / 1472 
SUBSISTEMA: usb 
TAGS:'assento' 
TYPE: 0/0/0 
USEC_INITIALIZED:' 2122311 '

**Comment by charles8119** [comment]
Then I use this comand:
ls -l /dev/serial/by-id

NO such file or directory.

Any suggestion?

**Comment by Twit123** [comment]
What does ls -al /dev/ttyA* and ls -al /dev/ttyU* give you?

Post results

Simon

**Reply by charles8119** [reply]
Simon, thank you so much for your support.

~ $  ls -al /dev/ttyA*
crw-rw----    1 root     audio     204,  64 Mar 17 15:06 /dev/ttyAMA0
~ $ 
~ $ 
~ $ 
~ $ ls -al /dev/ttyU*
ls: /dev/ttyU*: No such file or directory

**Comment by Twit123** [comment]
The user needs to be a member of the dialout group.

Mine looks like this
crw-rw---- 1 root dialout 166, 0 Mar 16 08:43 /dev/ttyACM0

I find it strange that the ttyAMA0 is a member of the audio group, is that not the built in audio?

If you remove the CC2531 and run lsusb then put the CC2531 back and run lsusb again, do you see the extra line where it was added?

Simon
